src/main/java/org/prebid/server/bidder/medianet/MedianetBidder.java

```diff
@@ -1,14 +1,99 @@
 package org.prebid.server.bidder.medianet;
 
-import org.prebid.server.bidder.OpenrtbBidder;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
 import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
 
-public class MedianetBidder extends OpenrtbBidder<Void> {
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class MedianetBidder implements Bidder<BidRequest> {
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
 
     public MedianetBidder(String endpointUrl, JacksonMapper mapper) {
-        super(endpointUrl,
-                RequestCreationStrategy.SINGLE_REQUEST,
-                Void.class,
-                mapper);
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        if (CollectionUtils.isEmpty(request.getImp())) {
+            return Result.withError(BidderError.badInput("No valid impressions in the bid request"));
+        }
```

Comment:
This is impossible condition(due to [this validation](https://github.com/prebid/prebid-server-java/blob/854f5b0f09e45aedf3d2f44cd4e7ca32108e688e/src/main/java/org/prebid/server/validation/RequestValidator.java#L147)), please, remove this check.